packages/bathymetry/src/bathy.maker.ts

```diff
@@ -0,0 +1,218 @@
+import { LogType } from '@basemaps/shared';
+import * as fs from 'fs';
+import PLimit from 'p-limit';
+import type { Limit } from 'p-limit';
+import * as path from 'path';
+import { File, FileType } from './file';
+import { Hash } from './hash';
+import { MapnikRender } from './mapnik';
+import { Stac } from './stac';
+
+interface BathyMakerContext {
+    path: string;
+    tms: TileMatrixSet;
+    zoom: number;
+    /** Number of threads used to convert @default 8 */
+    threads?: number;
+    /** Mapnik render tilesize @default 8192 */
+    tileSize?: number;
+}
+
+export class BathyMaker {
+    filePath: string;
+    tms: TileMatrixSet;
+    zoom: number;
+    q: Limit;
+    tileSize: number;
+    gdalVersion: Promise<string>;
+    file: File;
+
+    constructor(ctx: BathyMakerContext) {
+        this.filePath = ctx.path;
+        this.tms = ctx.tms;
+        this.zoom = ctx.zoom;
+        this.q = PLimit(ctx.threads || 8);
+        this.tileSize = ctx.tileSize || 8192;
+        this.file = new File(ctx.path);
+    }
+
+    /** Most operations need a tiff file */
+    get tiffPath(): string {
+        if (this.filePath.endsWith('.tiff')) return this.filePath;
+        return this.filePath + '.tiff';
+    }
+
+    /** File name of the path */
+    get fileName(): string {
+        return path.basename(this.filePath);
+    }
+
+    async render(logger: LogType): Promise<void> {
+        console.log(this.filePath);
+        this.gdalVersion = Gdal.version(logger);
+
+        const isNc = this.filePath.endsWith('.nc');
+
+        // NetCdf files need to be converted to GeoTiff before processing
+        if (isNc) await this.createSourceGeoTiff(logger);
+        await this.createSourceHash(logger);
+
+        const zoom = this.tms.zooms[this.zoom];
+        logger.info(
+            { file: this.fileName, zoom: this.zoom, tileCount: zoom.matrixWidth * zoom.matrixHeight },
+            'Splitting Tiles',
+        );
+
+        const gdalVersion = await this.gdalVersion;
+        logger.info({ version: gdalVersion }, 'GdalVersion');
+
+        const promises = [];
+        for (let x = 0; x < zoom.matrixWidth; x++) {
+            for (let y = 0; y < zoom.matrixHeight; y++) {
+                const promise = this.q(() => {
+                    const tile = { x, y, z: this.zoom };
+                    const tileId = TileMatrixSet.tileToName(tile);
+                    return this.renderTile(tile, logger.child({ index: tileId }));
+                });
+                promises.push(promise);
+            }
+        }
+        await Promise.all(promises).catch((err) => logger.fatal(err, 'FailedToRun'));
+    }
+
+    /** Create a multi hash of the source file  */
+    async createSourceHash(logger: LogType): Promise<string> {
+        const hashPath = this.file.name(FileType.Hash);
+        if (fs.existsSync(hashPath)) return (await fs.promises.readFile(hashPath)).toString();
+        logger.info({ hashPath }, 'CreateHash');
```

Comment:
Do we want to not re-create the hash if it exists already; maybe it changed?

Reply:
I was thinking about this, I could check a fs stat to see modified time or something, I think what ideally i want to do is delete the `.bathy` folder after a success